test/test_direct_dht.cpp

```diff
@@ -99,9 +99,9 @@ TORRENT_TEST(direct_dht_request)
 	sp.set_bool(settings_pack::enable_upnp, false);
 	sp.set_str(settings_pack::dht_bootstrap_nodes, "");
 	sp.set_int(settings_pack::max_retry_port_bind, 800);
-	sp.set_str(settings_pack::listen_interfaces, "127.0.0.1:42434");
+	sp.set_str(settings_pack::listen_interfaces, "0.0.0.0:42434");
 	lt::session responder(sp, {});
-	sp.set_str(settings_pack::listen_interfaces, "127.0.0.1:45434");
+	sp.set_str(settings_pack::listen_interfaces, "0.0.0.0:45434");
```

Comment:
Hrm, if we start expanding `0.0.0.0` these tests will break again.

Reply:
you're right. what is the right thing to do? maybe if one explicitly binds to loopback, it's OK to start a DHT node on it, as long as it stays there, and doesn't try to reach out to non-loopback addresses.

Reply:
on the other hand, if one binds to `0.0.0.0`, I think it's expected to be able to reach "the one IPv4" DHT node via `127.0.0.1`